src/arch/xtensa/lib/cpu.c

```diff
@@ -38,24 +38,69 @@ extern struct xtos_core_data *core_data_ptr[PLATFORM_CORE_COUNT];
 
 static uint32_t active_cores_mask;
 
+#if CONFIG_NO_SLAVE_CORE_ROM
+extern void *shared_vecbase_ptr;
+extern void _WindowOverflow4;
```

Comment:
Did you mean `extern void *_WindowOverflow4;` ? This fails to compile as soon as you try to do anything with it:

```
.../xtensa/lib/cpu.c:65:38: error: taking address of expression of type 'void' [-Werror]
  void *dyn_vec_start_addr = (void *)(&_WindowOverflow4);
```

Reply:
No, _WindowOverflow4 and it works fine

Reply:
Interesting, which compiler is this? AFAIK the C language doesn't define anything like this.

That error is with `gcc (crosstool-NG 1.23.0.440-3b8c) 8.1.0` built per the SOF documentation.

Tigerlake is not built by default by ./scripts/xtensa-build-all.sh, that's why CI doesn't fail like this. Yet.

Reply:
Im using only xcc.
Sometimes there are some issues with gcc, for example https://github.com/thesofproject/sof/issues/3035

Reply:
@abonislawski my expectation is that *all* platforms should build with GCC and this build test should be part of CI.
@marc-hb can you add TGL + GCC to the build script when time permits.

Reply:
> Im using only xcc.

The more compilers and analyzers are used and the more bugs are found. Sometimes they're bugs in compilers. Most of the time they're bugs in the compiled code.

There are plenty enough undefined behaviors with Standard C code and rarely ever the need to write non-standard code; I doubt it's the case here. is it?